sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ResourceThrottleRetryPolicy.java

```diff
@@ -86,6 +86,15 @@ public Mono<ShouldRetryResult> shouldRetry(Exception exception) {
         if (this.currentAttemptCount < this.maxAttemptCount &&
                 (retryDelay = checkIfRetryNeeded(dce)) != null) {
             this.currentAttemptCount++;
+
+            logger.warn(
```

Comment:
I am wondering whether logging the request resourceType and operationType would also help here - but did not see request maintained here anywhere, so will not be an quick change

Reply:
CosmosDiagnosticsContext is null so will require some wiring work - will keep it aside for now.